ee/modules/160-multitenancy-manager/templates/validating.yaml

```diff
@@ -0,0 +1,41 @@
+{{- if semverCompare ">= 1.26" .Values.global.discovery.kubernetesVersion }}
+---
+{{- if semverCompare ">= 1.28" .Values.global.discovery.kubernetesVersion }}
+apiVersion: admissionregistration.k8s.io/v1beta1
+{{- else }}
+apiVersion: admissionregistration.k8s.io/v1alpha1
+{{- end }}
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "multitenancy-manager"
+  {{- include "helm_lib_module_labels" (list . (dict "app" "multitenancy-manager") ) | nindent 2 }}
+spec:
+  failurePolicy: Fail
+  matchConstraints:
+    resourceRules:
+      - apiGroups:   ["*"]
+        apiVersions: ["*"]
+        operations:  ["UPDATE", "DELETE"]
+        resources:   ["*"]
+        scope: Namespaced
+  validations:
+    - expression: 'request.userInfo.username == "system:serviceaccount:d8-system:deckhouse"'
+      messageExpression: "'This resource is managed by ' + string(object.metadata.namespace) + ' Project. Manual modification is forbidden.'"
+---
+{{- if semverCompare ">= 1.28" .Values.global.discovery.kubernetesVersion }}
+apiVersion: admissionregistration.k8s.io/v1beta1
+{{- else }}
+apiVersion: admissionregistration.k8s.io/v1alpha1
+{{- end }}
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: "multitenancy-manager-binding"
+  {{- include "helm_lib_module_labels" (list . (dict "app" "multitenancy-manager") ) | nindent 2 }}
+spec:
+  policyName: "multitenancy-manager"
+  validationActions: [Deny, Audit]
+  matchResources:
+    namespaceSelector:
+      matchLabels:
+        d8-module: multitenancy-manager
```

Comment:
Does it mean that all resources can only be created by deckhouse?
It looks like we validate all resources in namespaces labeled by the `d8-module: multitenancy-manager`? 

It seems like a user cannot create anything in this namespaces, or am I wrong?

Reply:
It affects only `UPDATE` and `DELETE` operations. You still can create resources
```yaml
# kubectl -n test-project-1 create secret generic foobar
secret/foobar created
```

But after it, you will be unable to modify/delete them. 
Probably we can add some extra label for manual resources

Reply:
Changed. Description was updated